q2_phylogeny/_filter.py

```diff
@@ -19,3 +22,45 @@ def filter_table(table: biom.Table, tree: skbio.TreeNode) -> biom.Table:
     ids_to_keep = tip_ids & feature_ids
     table.filter(ids_to_keep, axis='observation', inplace=True)
     return table
+
+
+def filter_tree(tree: skbio.TreeNode,
+                table: biom.Table = None,
+                sequences: pd.Series = None,
```

Comment:
thanks @jwdebelius !

you only need metadata... QIIME 2 can automatically transform the sequences or table to a `Metadata` object, so accepting `FeatureData[Sequence]` is totally redundant.

Accepting a table is also a bit redundant, though it occurs to me that the table needs to be transposed for the IDs to consist of feature IDs, so also accepting a table could be kept as a convenience.

Reply:
I like the tree because I think as a user, I'd prefer the explicitness. I kind of liked the sequences because they were also explicit.